tests/jax/components/training/losses_test.py

```diff
@@ -0,0 +1,194 @@
+            "network_agent": SimpleNamespace(
+                network=MockNet, log_prob=log_prob, entropy=entropy
+            )
+        }
+    }
+
+    key = jax.random.PRNGKey(5)
+    action_info = "action_info_test"
+    policy_info = "policy_info_test"
+
+    parameters = {
+        "network_agent": {
+            "mlp/~/linear_0": {
+                "w": jnp.array([[0.0, 0.0, 0.0, 0.0]]),
+            }
+        }
+    }
+
+    store = SimpleNamespace(
+        is_evaluator=None,
+        observations=observations,
+        parameters=parameters,
+        trainer_agents=["agent_0", "agent_1", "agent_2"],
+        networks=network,
+        network=network,
+        agent_net_keys=agent_net_keys,
+        trainer_agent_net_keys={
+            "agent_0": "network_agent",
+            "agent_1": "network_agent",
+            "agent_2": "network_agent",
+        },
+        key=key,
+        action_info=action_info,
+        policy_info=policy_info,
+    )
+
+    mock_trainer = Trainer(store)
+
+    return mock_trainer
+
+
+@pytest.fixture
+def mapg_trust_region_clipping_loss() -> MAPGWithTrustRegionClippingLoss:
+    """Creates an MAPG loss fixture with trust region and clipping"""
+
+    test_mapg = MAPGWithTrustRegionClippingLoss(
+        config=MAPGTrustRegionClippingLossConfig(value_cost=0.5, entropy_cost=0.01)
+    )
+
+    return test_mapg
+
+
+def test_mapg_creation(
+    mock_trainer: Trainer,
+    mapg_trust_region_clipping_loss: MAPGWithTrustRegionClippingLoss,
+) -> None:
+    """Test whether mapg function is successfully created"""
+
+    mapg_trust_region_clipping_loss.on_training_loss_fns(trainer=mock_trainer)
+    assert hasattr(mock_trainer.store, "grad_fn")
+    assert isinstance(mock_trainer.store.grad_fn, Callable)  # type:ignore
+
+
+def test_mapg_loss(
+    mock_trainer: Trainer,
+    mapg_trust_region_clipping_loss: MAPGWithTrustRegionClippingLoss,
+) -> None:
+    """Test whether mapg loss output is as expected"""
+    mapg_trust_region_clipping_loss.on_training_loss_fns(trainer=mock_trainer)
+    grad_fn = mock_trainer.store.grad_fn
+
+    actions = {
+        agent: jnp.array([1.0, 1.0, 1.0, 1.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    behaviour_log_probs = {
+        agent: jnp.array([-2.0, -2.0, -2.0, -2.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    target_values = {
+        agent: jnp.array([3.0, 3.0, 3.0, 3.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    advantages = {
+        agent: jnp.array([2.0, 2.0, 2.0, 2.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    behavior_values = {
+        agent: jnp.array([1.0, 1.0, 1.0, 1.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+
+    low_target_values = {
+        agent: jnp.array([1.0, 1.0, 1.0, 1.0])
+        for agent in {"agent_0", "agent_1", "agent_2"}
+    }
+    low_advantages = {
+        agent: jnp.array([3.0, 3.0, 3.0, 3.0])
```

Comment:
Maybe set these to be all 1.0 to make them lower than `advantages`

Reply:
The naming is confusing I think? They're used to generate the lower losses so they need to be higher to create a lower `loss_total`.

Reply:
I see. Thank you for clarifying.